BeeTracker/app/src/main/java/com/blueprint/foe/beetracker/MainActivity.java

```diff
@@ -26,95 +28,159 @@
-                navigateToHome();
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                resetInputFieldActiveState(emailInput);
             }
+
+            @Override
+            public void afterTextChanged(Editable s) {}
         });
 
-        LoginButton loginButton = (LoginButton) findViewById(R.id.login_button);
-        loginButton.setReadPermissions("email");
-        // Other app specific specialization
+        final EditText passwordInput = (EditText) findViewById(R.id.password_input);
+        passwordInput.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {}
 
-        // Callback registration
-        callbackManager = CallbackManager.Factory.create();
-        loginButton.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onClick(View view) {
-                spinningIconDialog = new SpinningIconDialog();
-                spinningIconDialog.show(getFragmentManager(), "SpinningPopup");
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                resetInputFieldActiveState(passwordInput);
             }
+
+            @Override
+            public void afterTextChanged(Editable s) {}
         });
-        loginButton.registerCallback(callbackManager, new FacebookCallback<LoginResult>() {
+
+        Button emailLoginButton = (Button) findViewById(R.id.email_login_button);
+        emailLoginButton.setOnClickListener(new View.OnClickListener() {
             @Override
-            public void onSuccess(LoginResult loginResult) {
+            public void onClick(View view) {
+                spinningIconDialog = new SpinningIconDialog();
+                spinningIconDialog.show(getFragmentManager(), "SpinningPopup");
 
                 BeeTrackerCaller caller = new BeeTrackerCaller();
                 try {
-                    Call<BeeTrackerCaller.SignupResponse> token = caller.signup(loginResult.getAccessToken());
-                    token.enqueue(loginCallback);
+                    String email = ((EditText) findViewById(R.id.email_input)).getText().toString();
+                    String password = ((EditText) findViewById(R.id.password_input)).getText().toString();
+                    Call<BeeTrackerCaller.EmailPasswordSigninResponse> call = caller.emailPasswordSignin(email, password);
+                    call.enqueue(emailPasswordLoginCallback);
                 } catch (IOException e) {
                     Log.e(TAG, e.toString());
                     e.printStackTrace();
                     showErrorDialog(getString(R.string.error_message_login));
+                    spinningIconDialog.dismiss();
```

Comment:
Can we move this into showErrorDialog? Also please guard the dismiss call by checking if spinningIconDialog is null. We're having an issue with null pointers in other places in the app and will be implementing this check throughout.